File: python3/src/models/player.py
from models.gameObject import GameObject
from models.bomb import Bomb

class Player(GameObject):
    # playerId: PlayerId;
    # actions: PlayerAction;
    # /** The amount of pixels that the player moves at every game tick */
    # speed: number;
    # /** Maximum number of bombs that the player can drop at a time. */
    # maxBombCount: number;
    # /** All the bombs currently in the map that were dropped by the player */
    # bombs: Bomb[];
    # isAlive: boolean;
    pass


# Moves that the player can do.
class PlayerAction:
    # move_up = False
    # move_down = False
    # move_left = False
    # move_right = False
    # plant_bomb = False
    pass

def player_action_factory():
    action = {}
    action["move_up"] = False
    action["move_down"] = False
    action["move_left"] = False
    action["move_right"] = False
    action["plant_bomb"] = False
    return action

File: python3/src/models/player.ts
import { GameObject } from "./game-object";
import { Bomb } from "./bomb";

export type PlayerId = string;

export interface Player extends GameObject {
    playerId: PlayerId;
    /** The inputs of this player. */
    actions: PlayerAction;
    /** The amount of pixels that the player moves at every game tick */
    speed: number;
    /** Maximum number of bombs that the player can drop at a time. */
    maxBombCount: number;
    /** All the bombs currently in the map that were dropped by the player */
    bombs: Bomb[];
    isAlive: boolean;

    /**
     * The power of the explosion for this bomb, in map cell.
     * This power is the number of cells that will be affected by the bomb,
     * starting from the position of the bomb to the end of one side of the explosion.
     */
    bombPower:number;
}

// Moves that the player can do.
export class PlayerAction {
    move_up = false;
    move_down = false;
    move_left = false;
    move_right = false;
    plant_bomb = false;
}

/** Used to describe the action of a given player for the socket communication */
export interface PlayerActionWrapper {
    playerId: PlayerId;
    actions: PlayerAction;
}
